src/trace.rs

```diff
@@ -0,0 +1,212 @@
+// For completeness, wrappers around all of tracing's public macros are provided, even if they are
+// not used at the present time.
+#![allow(unused_macros)]
+
+//! Internal Tracing macro module
+//!
+//! The [`trace`][crate::trace] module is an internal module that contains wrapper macros encapsulating
+//! [`tracing`][tracing]'s macros. These macros allow for conditional expansion of
+//! [`tracing`][tracing] macros during compilation when the `tracing` feature is enabled, or trimming
+//! them when the feature is disabled, all in a concise manner.
+//!
+//! The macros from the [`trace`][crate::trace] module are declared by default and can be used throughout the
+//! crate. However, as the contents of these macros are conditionally compiled, they are effectively trimmed
+//! during inline expansion when the `tracing` feature is disabled.
+//!
+//! # Unstable
```

Comment:
This makes me realize that in `lib.rs`, there isn't a section about unstable features. We should move the essence of this there (without describing what the internals of this module does).